src/java.base/share/classes/java/lang/Long.java

```diff
@@ -311,7 +313,36 @@ private static BigInteger toUnsignedBigInteger(long i) {
      * @since   1.0.2
      */
     public static String toHexString(long i) {
-        return toUnsignedString0(i, 4);
+        int mag = Long.SIZE - Long.numberOfLeadingZeros(i);
+        int len = Math.max(((mag + 3) / 4), 1);
+        byte coder = COMPACT_STRINGS ? LATIN1 : UTF16;
+        byte[] chars = new byte[len << coder];
+        byte b;
+        long x = Integer.hex8(i);
+        if (len > 8) {
+            if (COMPACT_STRINGS) {
+                len -= 8;
+                Unsafe.getUnsafe().putLong(chars, ARRAY_BYTE_BASE_OFFSET + len, Long.reverseBytes(x));
```

Comment:
We only need to reverse on small endian platforms right? We can use putLongUnaligned which takes a boolean for big endian so conversion is automatic.